code/dao/base_dao.py

```diff
@@ -0,0 +1,62 @@
+from typing import TypedDict, Optional
+
+from config.db import DBManager
+
+
+class ListParams(TypedDict, total=False):
+    sort: Optional[str]
+    query: Optional[str]
+    page: Optional[int]
+    limit: Optional[int]
+
+
+class BaseDAO:
+
+    def __init__(self, table: str, db_url: str):
+        self._db = DBManager(db_url)
+        self._cur = self._db.connection.cursor()
```

Comment:
En general, debes desacoplar el acceso a datos de los que consuman esos datos, usando inyección de dependencias. Los que accedan a los datos no deben tener ninguna pista sobre cómo se implementa el mismo (si con base de datos, o cómo). De hecho, en un diseño por capas debes dejar el acceso a datos para el final.

Reply:
removed (for now) all these stuff...